libs/scripts/personalization.js

```diff
@@ -0,0 +1,406 @@
+/* eslint-disable no-console */
+let utils;
+
+const CLASS_EL_DELETE = 'p13n-deleted';
+const CLASS_EL_REPLACE = 'p13n-replaced';
+
+export const PERSONALIZATION_TAGS = {
+  chrome: () => navigator.userAgent.includes('Chrome') && !navigator.userAgent.includes('Mobile'),
+  firefox: () => navigator.userAgent.includes('Firefox') && !navigator.userAgent.includes('Mobile'),
+  android: () => navigator.userAgent.includes('Android'),
+  ios: () => /iPad|iPhone|iPod/.test(navigator.userAgent),
+  loggedout: () => !window.adobeIMS?.isSignedInUser(),
+  loggedin: () => window.adobeIMS?.isSignedInUser(),
+  darkmode: () => window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches,
+  lightmode: () => !PERSONALIZATION_TAGS.darkmode(),
```

Comment:
wouldn't this throw a lint error - `PERSONALIZATION_TAGS was used before it was defined`?!

Reply:
The alternative is:
`lightmode: function () { return !this.darkmode(); },`

but I prefer the current one.  Thoughts?